### PR TITLE
minor fixes at import storage domain

### PR DESCRIPTION
**Pull request by maorlipchuk**
Fixing video links and removing BRs from import storage domain wiki

### CODE REVIEW

File: source/develop/release-management/features/storage/importstoragedomain.html.md [hunk 1]
```diff
@@ -393,6 +393,37 @@ If the user want to register a specific floating disks in the system he should u
...
+    The volumes might be missing from the following reasons:
+    1. Not all storage domains were imported to the engine, and some of the VM's/Template's disks were dependant on this storage domains.
+    2. The OVF_STORE disk was not synced when the disaster occured, causing the volumes in the VM/Template to be unsynced.
+    This might happen in numerous scenarios such as remove a snapshot, delete a disk or any other operation which changed the volume chain.
+
+    This type of failure will be logged in the engine, for example:
+
+    <I>ERROR [org.ovirt.engine.core.vdsbroker.irsbroker.GetImageInfoVDSCommand] (default task-47) [14de7d5f] IrsBroker::getImageInfo::Failed getting image info imageId='425daada-2d07-4d7f-9365-4674e2ef50a0' does not exist on domainName='data4', domainId='d07e18b6-20da-46b8-a7e5-5688dc3cb04a', error code: 'ImagePathError', message: Image path does not exist or cannot be accessed/created: (u'/rhev/data-center/mnt/10.35.16.43:_export_data4/d07e18b6-20da-46b8-a7e5-5688dc3cb04a/images/75a157ee-c485-423d-9c0e-62d5d3b9d718',)</I>
```
Comment (garrett) [inline source/develop/release-management/features/storage/importstoragedomain.html.md]: This should probably be in a quote block, like so:

> ERROR [org.ovirt.engine.core.vdsbroker.irsbroker.GetImageInfoVDSCommand] (default task-47) [14de7d5f] IrsBroker::getImageInfo::Failed getting image info imageId='425daada-2d07-4d7f-9365-4674e2ef50a0' does not exist on domainName='data4', domainId='d07e18b6-20da-46b8-a7e5-5688dc3cb04a', error code: 'ImagePathError', message: Image path does not exist or cannot be accessed/created: (u'/rhev/data-center/mnt/10.35.16.43:_export_data4/d07e18b6-20da-46b8-a7e5-5688dc3cb04a/images/75a157ee-c485-423d-9c0e-62d5d3b9d718',)

(Basically, start the line with a `>`. You're not quoting someone, but you are quoting some<em>thing</em>.)

**Comment by mureinik** [comment]
:+1:

**Comment by garrett** [comment]
@maorlipchuk: I've made a bunch of changes in a PR on your fork. When you accept the PR on your own fork, the commits should also show up in this PR here.

I guess the next step is for us to start codifying a style guide based on the changes the commits so it's easier for everyone to know what things to change to make documents a bit more optimal. :wink:

**Comment by maorlipchuk** [comment]
Thanks! great job!  :)

Would you want to add a new page, describing the standards in the wiki,
maybe even link it as part of the README file.

I guess we should add there something like this :

1. Bullet should be formatted in the wiki using a star at the beginning of
the line and a single space after it as follow :
 "* Example"
2. Links -  A link should be formatted with a square bracket, indicating
the desired readable name to be used in the wiki page, and following the
local path which the page should be found, for example:
[ImportUnregisteredEntities](ImportUnregisteredEntities).

** BTW, I've just notice that some of the links are broken as part of
change d16b7563cbec01cc4fb29f42664be6d2e4fe21cd

links to importunregisteredentities page which were directed to
/features/engine/importunregisteredentities/
is now directing to
/features/storage/importstoragedomain/ImportUnregisteredEntities
Perhaps, it will be better to also change the location of
ImportUnregisteredEntities to be under the storage folder.

2. Workflow - If your wiki include workflows describing how the feature
should be operated, please describe the actions in a short and precise way
using numerable steps in the following format [Number][dot][Space], or
example:

  1. Initialize...
  2. Press ....
  3. ...

*Other comments*

1. Avoid using <br> html element, the browser should handle this.
2.  gender-specific pronouns - When describing a scenario use plural
reference (they is better than one/he/she).
3. Use kbd/code type for the following keywords:
  VM
  Template
  oVirt
  Ceph
  LUN
  Data Center


*Tests that can be added:*
*capitalized initialisms* - We can add a test which will go over the source
and search for " vm " or " template " to indicate if those should be
capitalized.

*Links validity* - Add a script which will go over the source code and
check for all the links validity - If we get 500 or 404 then this link is
not valid.
That way, if a youtube video/wiki page will be moved or deleted and other
wiki pages will use it as part of their links, a test can notify about it
automatically.

Maybe we should also add a standard for commit messages, for example, we
should indicate the page name in the commit message, like :
 "importstoragedomain: optimized code blocks"

Thanks,
Maor

**Comment by maorlipchuk** [comment]
On Sun, May 22, 2016 at 11:27 AM, Maor Lipchuk <mlipchuk@redhat.com> wrote:

> Thanks! great job!  :)
>
> Would you want to add a new page, describing the standards in the wiki,
> maybe even link it as part of the README file.
>
> I guess we should add there something like this :
>
> 1. Bullet should be formatted in the wiki using a star at the beginning of
> the line and a single space after it as follow :
>  "* Example"
> 2. Links -  A link should be formatted with a square bracket, indicating
> the desired readable name to be used in the wiki page, and following the
> local path which the page should be found, for example:
> [ImportUnregisteredEntities](ImportUnregisteredEntities).
>
> ** BTW, I've just notice that some of the links are broken as part of
> change d16b7563cbec01cc4fb29f42664be6d2e4fe21cd
>
> links to importunregisteredentities page which were directed to
> /features/engine/importunregisteredentities/
> is now directing to
> /features/storage/importstoragedomain/ImportUnregisteredEntities
> Perhaps, it will be better to also change the location of
> ImportUnregisteredEntities to be under the storage folder.
>
> 2. Workflow - If your wiki include workflows describing how the feature
> should be operated, please describe the actions in a short and precise way
> using numerable steps in the following format [Number][dot][Space], or
> example:
>
>   1. Initialize...
>   2. Press ....
>   3. ...
>
> *Other comments*
>
> 1. Avoid using <br> html element, the browser should handle this.
> 2.  gender-specific pronouns - When describing a scenario use plural
> reference (they is better than one/he/she).
> 3. Use kbd/code type for the following keywords:
>   VM
>   Template
>   oVirt
>   Ceph
>   LUN
>   Data Center
>
>
> *Tests that can be added:*
> *capitalized initialisms* - We can add a test which will go over the
> source and search for " vm " or " template " to indicate if those should be
> capitalized.
>
> *Links validity* - Add a script which will go over the source code and
> check for all the links validity - If we get 500 or 404 then this link is
> not valid.
> That way, if a youtube video/wiki page will be moved or deleted and other
> wiki pages will use it as part of their links, a test can notify about it
> automatically.
>


I've worked on a script which go over all the resources in the ovirt-site
project and search for the following prefix [name](site link).
Once it found that it validate if the link is valid.

For now it only works for "http://" sites, but it already found some
invalid links that should be fixed.

I'm adding the script to the mail and a list of invalid URLs which it found.

I will upload thos script to github once I will verify it works well and
hopefully we can integrate this with the ovirt-site as a test.

**Comment by garrett** [comment]
Oh! I think the links that are broken are the ones containing hashes... I would guess that there's a bug in my special wiki-handling link code. I'll look into this today.

**Comment by maorlipchuk** [comment]
Sure, thanks!
I found around 270 potential broken http links, I've attached a file
describing them all.
Most of the links does not contains hashes, it uses httplib package and
validate the HEAD of the request for getting a valid response code.
If the response code is 404 or 500 then it indicates it is a broken link.

I've opened a new repo for this utility, it might be valuable for other
wiki projects as well, feel free to take a look:
  https://github.com/maorlipchuk/WikiLinksValidator

If you need any help fixing them please let me know, maybe we can use this
script, or changing your script so it can be an automatic test (Although I
would run it every day instead of every commit since the validation takes
about 30 minutes each time)

Regards,
Maor

**Comment by garrett** [comment]
While digging around, I found that redirects are not properly handled in the code itself, but relies on Apache redirect rules (and hopefully those would work). I'm rewriting the wiki link finding function to handle redirects at the moment, as well as trying to make sure it works with hashes in your document.

Some of the broken links you might be getting in testing may be fixed via Apache redirects.

**Comment by maorlipchuk** [comment]
yes, I guess you are right, it will probably fix many failures.

I see that many broken links are using "category" like:
http://www.ovirt.org/Category:Feature
http://www.ovirt.org/Category:TestCase
http://www.ovirt.org/Category:OVirt_3.6_Feature

Those links should be re-directed as so:
http://old.ovirt.org/Category:Feature
http://old.ovirt.org/Category:TestCase
http://old.ovirt.org/Category:OVirt_3.6_Feature

Regards,
Maor

**Comment by mureinik** [comment]
@maorlipchuk Can you please rebase this branch so it can be merged?

**Comment by sandrobonazzola** [comment]
Please rebase

**Comment by maorlipchuk** [comment]
Hi,

I can't seem to push my rebase to the ovirt-site repo because I don't have the correct access rights.
I've made a different pull request at https://github.com/oVirt/ovirt-site/pull/336 which is the rebase of this PR.
Please let me know if the rebase should be handled differently, or that the new PR is enough to work with, and I will close this one

**Comment by mureinik** [comment]
Closing, since #336 is merged.

**Pull request closed without merging**